src/java.base/share/classes/java/security/spec/MGF1ParameterSpec.java

```diff
@@ -162,4 +162,9 @@ public MGF1ParameterSpec(String mdName) {
     public String getDigestAlgorithm() {
         return mdName;
     }
+
+    @Override
+    public String toString() {
+        return "MGF1:" + mdName;
```

Comment:
The current MGF part shows `MGF: MGF1:SHA-256`. An extra space would looks a little strange, and you don't know if the key is `MGF: MGF1` or ``MGF`. I'd rather keep the current output.

Reply:
How about just MGF: MGF1SHA-256 or MGF: MGF1 SHA-256?
First combines into a single string, the second uses space instead of ":". Just a suggestion.

Reply:
I like the 2nd one better, but I'll see how Sean thinks about the verbose record-style toString.

Reply:
99% of the time, it will probably be one of the MGF1ParameterSpec constants. For those, perhaps it would be nice to just use the constant: "MGF: MGF1ParameterSpec.SHA1" and for anything else, "MGF: MGF1ParameterSpec [digestAlgorithm: ...>". If this is too complicated, then I am ok with the above.

Reply:
This is doable and probably will be fast if I just keep checking `if (this == MGF1ParameterSpec.SHA1) return "MGF1ParameterSpec.SHA1"` but the code will be quite long. There are 11 constants. Maybe if this class was an enumeration...